src/Sentry/Http/HttpTransportBase.cs

```diff
@@ -397,6 +425,26 @@ await envelope
             }
         }
 
+        private void IncrementDiscardsForHttpFailure(HttpStatusCode responseStatusCode, Envelope envelope)
+        {
+            // First, restore any counts that were trying to be sent, so they are not lost.
+            var clientReportItems = envelope.Items.Where(x => x.TryGetType() == "client_report");
+            foreach (var item in clientReportItems)
+            {
+                var clientReport = (ClientReport)((JsonSerializable)item.Payload).Source;
+                _options.ClientReportRecorder.Load(clientReport);
+            }
+
+            if ((int)responseStatusCode is 429 or < 400)
+            {
+                //  Status == 429 or < 400 should not be counted by the client SDK
+                //  See https://develop.sentry.dev/sdk/client-reports/#sdk-side-recommendations
+                return;
```

Comment:
if we return here, isnt the `_options.ClientReportRecorder.Load(clientReport);` above redundant?

Reply:
I suppose that depends on if the server accepted the client report, even though it rate limited the other items in the envelope.  I wouldn't think so, but maybe @philipphofmann might have an idea about that?

Reply:
perhaps it just needs a comment to clarify?

Reply:
Got an answer internally, and we updated the spec here https://github.com/getsentry/develop/pull/568.  Will fix shortly.